src/app/static/src/js/components/reports/runReport.vue

```diff
@@ -0,0 +1,145 @@
+            </div>
+            <div v-on:click="dismissRunStatus" id="run-report-dismiss" class="btn btn-link p-0">Dismiss</div>
+        </div>
+    </div>
+</template>
+
+<script>
+    import {reportVersionToLongTimestamp} from "../../helpers";
+    import {session} from "../../session";
+    import axios from "axios";
+    export default {
+        name: 'runReport',
+        props: ['report'],
+        data() {
+            return {
+                showModal: false,
+                pollingTimer: null,
+                runningKey: "",
+                runningStatus: "",
+                newVersionFromRun: "",
+                newVersionDisplayName: ""
+            }
+        },
+        mounted() {
+            //check if we already have a runningStatusReport for this report in the session, and start monitoring it
+            const existingStatus = session.getRunningReportStatus(this.report.name);
+
+            this.runningStatus = existingStatus.runningStatus;
+            this.runningKey = existingStatus.runningKey;
+            this.newVersionFromRun = existingStatus.newVersionFromRun;
+            this.newVersionDisplayName = existingStatus.newVersionDisplayName;
+
+            if (this.runningStatus && !this.runHasCompleted()) {
+                this.startPolling();
+            }
+        },
+        methods: {
+            confirmRun: function () {
+                this.showModal = true;
+            },
+            cancelRun: function () {
+                this.showModal = false;
+            },
+            run: function () {
+                this.showModal = false;
+                axios.post(`/reports/${this.report.name}/run/`,
+                    null,
+                    {withCredentials: true})
+                    .then((response) => {
+                        this.runningKey = response.data.data.key;
+                        this.runningStatus = "Run started";
+
+                        this.startPolling();
+                    })
+                    .catch(() => {
+                        this.dismissRunStatus();
+                        this.runningStatus = "Error when running report";
+                    })
+                    .finally(() => {
+                        this.updateSessionStorage();
+                });
+            },
+            startPolling: function () {
+                if (this.pollingTimer) {
+                    this.stopPolling();
+                }
+
+                this.pollingTimer = setInterval(() => {
+                        axios.get(`/reports/${this.runningKey}/status/`,
+                            {withCredentials: true})
+                            .then((response) => {
+                                this.runningStatus = response.data.data.status;
+                                this.newVersionFromRun = response.data.data.version;
+                                this.newVersionDisplayName = this.newVersionFromRun ?
```

Comment:
Looks like `newVersionDisplayName` is a good candidate for a computed property,  also then you won't need to store it in session or worry about setting it in multiple places. Could also change `runHasCompleted` from a method to a computed and handle stopping polling inside of it, i.e.
```
runHasCompleted: function() {
    const completed = this.runningStatus === "success" || this.runningStatus === "error"
    if (completed){
              this.stopPolling();
    }
}
```
Then this callback gets a bit less complicated.